src/EditorFeatures/Core/Implementation/BlockCommentEditing/AbstractBlockCommentEditingCommandHandler.cs

```diff
@@ -2,76 +2,30 @@
 
 using System;
 using Microsoft.CodeAnalysis.Editor.Commands;
-using Microsoft.CodeAnalysis.Editor.Shared.Extensions;
-using Microsoft.CodeAnalysis.Editor.Shared.Options;
-using Microsoft.VisualStudio.Text;
 using Microsoft.VisualStudio.Text.Operations;
-using Roslyn.Utilities;
 
 namespace Microsoft.CodeAnalysis.Editor.Implementation.BlockCommentEditing
 {
-    internal abstract class AbstractBlockCommentEditingCommandHandler : ICommandHandler<ReturnKeyCommandArgs>
+    internal abstract class AbstractBlockCommentEditingCommandHandler : BaseAbstractBlockCommentEditingCommandHandler, 
+        ICommandHandler<ReturnKeyCommandArgs>
     {
-        private readonly ITextUndoHistoryRegistry _undoHistoryRegistry;
-        private readonly IEditorOperationsFactoryService _editorOperationsFactoryService;
-
         protected AbstractBlockCommentEditingCommandHandler(
             ITextUndoHistoryRegistry undoHistoryRegistry,
             IEditorOperationsFactoryService editorOperationsFactoryService)
+            : base(undoHistoryRegistry, editorOperationsFactoryService)
         {
-            Contract.ThrowIfNull(undoHistoryRegistry);
-            Contract.ThrowIfNull(editorOperationsFactoryService);
-
-            _undoHistoryRegistry = undoHistoryRegistry;
-            _editorOperationsFactoryService = editorOperationsFactoryService;
         }
 
         public CommandState GetCommandState(ReturnKeyCommandArgs args, Func<CommandState> nextHandler) => nextHandler();
 
         public void ExecuteCommand(ReturnKeyCommandArgs args, Action nextHandler)
         {
-            if (TryHandleReturnKey(args))
+            if (TryHandleReturnKey(args.SubjectBuffer, args.TextView))
             {
                 return;
             }
 
             nextHandler();
         }
-
-        private bool TryHandleReturnKey(ReturnKeyCommandArgs args)
```

Comment:
where did this code go?

Reply:
AbstractBlockCommentEditingCommandHandler is used by TypeScript so I cannot change it. So I refactored it into BaseAbstractBlockCommentEditingCommandHandler and PlatformAbstractBlockCommentEditingCommandHandler (not happy about that name though). Once TypeScript migrates to the new commanding, I will simplify it back.

Reply:
Never mind, it's better to have this class implement both ICommandHandler interfaces now and once TypeScript migrates over I can just delete legacy part.

---
In reply to: [159565154](https://github.com/dotnet/roslyn/pull/23769#discussion_r159565154) [](ancestors = 159565154)